torchbenchmark/models/timm_regnet/__init__.py

```diff
@@ -58,19 +64,21 @@ def set_eval(self):
         pass
 
     def _step_eval(self):
-        output = self.eval_model(self.cfg.infer_example_inputs)
+        output = self.eval_model(self.infer_example_inputs)
 
     def get_module(self):
-        return self.model, (self.cfg.example_inputs,)
+        self.example_inputs = self.example_inputs.to(device=self.device)
```

Comment:
Do runs include the data transfer time? I notice in the profiles multiple streams on GPU, such as Regnest14d eval streams 14 and 15, are those memory copies h2d?

Reply:
In fastNLP_Bert, the data transfer time is also included in train/eval. I guess it is more realistic to move the data in train/eval than prefetching them in init?

Reply:
My view is that doing copy right before use is recipe to create data stall. Correct user will prefetch, for example, next data, while using current loaded data. I don't ever see this done in oss pytorch. The reason we shouldn't do the load right before use is that it will create a bubble and make benchmark less sensitive.

We've talked for a while about how to model throughput and a big part of it is getting data copy done correctly.

I'm not sure what the right approach is, it might be to explicitly generate and copy next data while executing on previously loaded data?

I think for now, since we don't have a good answer, it might be best to just leave the copy in init and focus on compute time.

Reply:
I think for models that use simulated datasets, we try to simulate how it is used in the "wild" and do whatever model does for the task (such as fastNLP_Bert). For models that directly use generated tensors (and don't use dataloader), we preload data in __init__ and avoid bubbles.